packages/devtools_app/lib/src/network/network_controller.dart

```diff
@@ -384,49 +384,53 @@ class NetworkController
     } else {
-      filteredData.value = _requests.value.requests.where((NetworkRequest r) {
-        final methodArg = filter.filterArguments[methodFilterId];
-        if (methodArg != null &&
-            !methodArg.matchesValue(r.method.toLowerCase())) {
-          return false;
-        }
+      filteredData
+        ..clear()
+        ..addAll(_requests.value.requests.where((NetworkRequest r) {
+          final methodArg = filter.filterArguments[methodFilterId];
+          if (methodArg != null &&
+              !methodArg.matchesValue(r.method.toLowerCase())) {
+            return false;
+          }
 
-        final statusArg = filter.filterArguments[statusFilterId];
-        if (statusArg != null &&
-            !statusArg.matchesValue(r.status?.toLowerCase())) {
-          return false;
-        }
+          final statusArg = filter.filterArguments[statusFilterId];
+          if (statusArg != null &&
+              !statusArg.matchesValue(r.status?.toLowerCase())) {
+            return false;
+          }
 
-        final typeArg = filter.filterArguments[typeFilterId];
-        if (typeArg != null && !typeArg.matchesValue(r.type.toLowerCase())) {
-          return false;
-        }
+          final typeArg = filter.filterArguments[typeFilterId];
+          if (typeArg != null && !typeArg.matchesValue(r.type.toLowerCase())) {
+            return false;
+          }
 
-        if (filter.substrings.isNotEmpty) {
-          for (final substring in filter.substrings) {
-            final caseInsensitiveSubstring = substring.toLowerCase();
-            bool matches(String stringToMatch) {
-              if (stringToMatch
-                  .toLowerCase()
-                  .contains(caseInsensitiveSubstring)) {
-                _checkForError(r);
-                return true;
+          if (filter.substrings.isNotEmpty) {
+            for (final substring in filter.substrings) {
+              final caseInsensitiveSubstring = substring.toLowerCase();
+              bool matches(String stringToMatch) {
+                if (stringToMatch
+                    .toLowerCase()
+                    .contains(caseInsensitiveSubstring)) {
+                  _checkForError(r);
+                  return true;
+                }
+                return false;
               }
-              return false;
-            }
 
-            if (matches(r.uri)) return true;
-            if (matches(r.method)) return true;
-            if (matches(r.status ?? '')) return true;
-            if (matches(r.type)) return true;
+              if (matches(r.uri)) return true;
+              if (matches(r.method)) return true;
+              if (matches(r.status ?? '')) return true;
+              if (matches(r.type)) return true;
+            }
+            return false;
           }
-          return false;
-        }
-        _checkForError(r);
-        return true;
-      }).toList();
+          _checkForError(r);
+          return true;
+        }).toList());
```

Comment:
the `toList()`
isn't needed.

Reply:
without toList() an error like `'WhereIterable<LogData>' is not a subtype of type 'List<LogData>'` is thrown